src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -1587,9 +1587,8 @@ def get_boot_log(cmd, resource_group_name, vm_name):
     # Managed storage
     if blob_uri is None:
         try:
-            poller = client.virtual_machines.retrieve_boot_diagnostics_data(resource_group_name, vm_name)
-            uris = LongRunningOperation(cmd.cli_ctx)(poller)
-            blob_uri = uris.serial_console_log_blob_uri
+            boot_diagnostics_data = client.virtual_machines.retrieve_boot_diagnostics_data(resource_group_name, vm_name)
```

Comment:
is it because server doesn't return LROPoller in new api version? curious about what is the response type in previous api version.

Reply:
This is the earliest supported api-version, and the return object is `RetrieveBootDiagnosticsDataResult` [code link](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/compute/azure-mgmt-compute/azure/mgmt/compute/v2020_06_01/operations/_virtual_machines_operations.py#L1985)

By the way, when this logic was added before, there was no related test were added (the only legacy test was blocked).
https://github.com/Azure/azure-cli/pull/15145 So I assume this logic hasn't been tested before.